### PR TITLE
Merchant dashboard 

### PR DESCRIPTION
**Pull request by whitneykidd**
Completes user stories 34 - 37

### CODE REVIEW

File: app/controllers/admin/merchant_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  before_action :require_admin
+
+  def show
+    # binding.pry
+    @merchant = Merchant.find(params[:id])
+  end
+
+  private 
+    def require_admin
+      render file: "/public/404" unless current_admin?
```
Comment (whitneykidd) [inline app/controllers/admin/merchant_controller.rb:11]: Still need to create this view - will happen in future stories

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -2,14 +2,19 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
   helper_method :cart
-  helper_method :current_user
+  helper_method :current_user, :current_admin?
```
Comment (whitneykidd) [inline app/controllers/application_controller.rb:5]: Added a .current_admin? helper method that goes along with the '/public/404' functionality

File: app/models/user.rb [hunk 3]
```diff
@@ -7,7 +7,7 @@ class User < ApplicationRecord
   validates_confirmation_of :password
   has_many :orders
 
-
+  belongs_to :merchant, optional: true
```
Comment (whitneykidd) [inline app/models/user.rb:10]: This is how we can connect a merchant employee(a user with a role: 1) to the merchant it works for

File: spec/features/merchants/dashboard/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+RSpec.describe "Merchant Dashboard Page" do
+  describe "As a merchant, when I view my dashboard" do
+
+    it "I can see the name and full address of merchant I work for" do
```
Comment (whitneykidd) [inline spec/features/merchants/dashboard/show_spec.rb:6]: I couldn't get a before(:each) action to work in this file - would like to look more into this during a refactor

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -2,23 +2,36 @@
...
+    it {should validate_presence_of(:email)}
+    it {should validate_presence_of(:password)}
   end
 
-  describe "Relationships" do
+  describe "relationships" do
     it {should have_many :orders}
   end
+
+  describe "roles" do
```
Comment (whitneykidd) [inline spec/models/user_spec.rb:18]: Added these roles tests for role creation

File: spec/features/merchants/dashboard/show_spec.rb (hunk 4 above)
Comment (jhgould) [reply spec/features/merchants/dashboard/show_spec.rb:6]: I was having issues with before :each as well.

File: app/models/user.rb (hunk 3 above)
Comment (EricLarson2020) [reply app/models/user.rb:10]: Nice work

File: spec/models/user_spec.rb (hunk 5 above)
Comment (EricLarson2020) [reply spec/models/user_spec.rb:18]: Nice work here

**Pull request merged**